webrender/src/prim_store.rs

```diff
@@ -361,32 +356,46 @@ impl GradientData {
     }
 
     // Build the gradient data from the supplied stops, reversing them if necessary.
-    fn build(&mut self, src_stops: &[GradientStop], reverse_stops: bool) {
-        let mut cur_idx = 0usize;
-        let mut cur_color = if let Some(src) = src_stops.first() {
-            src.color
-        } else {
-            ColorF::new(0.0, 0.0, 0.0, 0.0)
-        };
+    fn build(&mut self, mut src_stops: AuxIter<GradientStop>, reverse_stops: bool) {
+
+        const MAX_IDX: usize = GRADIENT_DATA_RESOLUTION;
+        const MIN_IDX: usize = 0;
+
+        // Preconditions (should be ensured by DisplayListBuilder): 
+        // * we have at least two stops
+        // * first stop has offset 0.0
+        // * last stop has offset 1.0
```

Comment:
I did; the new debug_asserts and unwrap that imply these assumptions. I just wrote the actual assumption out to explain those asserts/unwrap.